### DISCUSSION TITLE
Reinventing workspaces

### DISCUSSION THREAD

**Discussion by blevkovych** [ergomake/layerform#23] [General]
Why don't I just use the terraform workspace?
https://developer.hashicorp.com/terraform/language/state/workspaces

**Comment by singlewind** [comment]
I don't know what you were refer to. I believe this resolve a different problem. A way to manage infrastructure as code in a hierarchical way. Can be wrong. Definitely will jump deep into it. Of cause, we can use terraform in hierarchical way. However, enforce a pattern can save some maintenance effort.

**Comment by lucasfcosta** [comment]
Terraform workspaces solve a different problem than this.

When using workspaces, you'll have an entirely separate state for everything you're applying. With Layerform, we break down state into layers, meaning you will be able to manage separate pieces of state more easily and componentize them.

I.e. you won't have to create a whole new stack to re-deploy the same pods on top of K8S Kubernetes.

Also, state management is another thing you can do with the CLI that you can't easily do with Terraform.

Just try to create usable staging environments with plain Terraform and I'm sure you'll find plenty of differences.
